doc/connectivity/usb/device_next/usb_device.rst

```diff
@@ -1,57 +1,187 @@
+configuration ``-DCONF_FILE=usbd_next_prj.conf`` either directly or via
+``west``.
+
+* :ref:`bluetooth-hci-usb-sample`
+
+* :zephyr:code-sample:`usb-cdc-acm`
+
+* :zephyr:code-sample:`usb-cdc-acm-console`
+
+* :zephyr:code-sample:`usb-mass`
+
+* :zephyr:code-sample:`usb-hid-mouse`
+
+* :zephyr:code-sample:`zperf` To build the sample for the new device support,
+  set the configuration overlay file
+  ``-DDEXTRA_CONF_FILE=overlay-usbd_next_ecm.conf`` and devicetree overlay file
+  ``-DDTC_OVERLAY_FILE="usbd_next_ecm.overlay`` either directly or via ``west``.
+
+How to configure and enable USB device support
+**********************************************
+
+For the USB device support samples in the Zephyr project repository, we have a
+common file for instantiation, configuration and initialization,
+:zephyr_file:`samples/subsys/usb/common/sample_usbd_init.c`. The following code
+snippets from this file are used as examples. USB Samples Kconfig options used
+in the USB samples and prefixed with ``SAMPLE_USBD_`` have default values
+specific to the Zephyr project and the scope is limited to the project samples.
+In the examples below, you will need to replace these Kconfig options and other
+defaults with values appropriate for your application or hardware.
+
+The USB device stack requires a context structure to manage its properties and
+runtime data. The preferred way to define a device context is to use the
+:c:macro:`USBD_DEVICE_DEFINE` macro. This creates a static
+:c:struct:`usbd_context` variable with a given name. Any number of contexts may
+be instantiated. A USB controller device can be assigned to multiple contexts,
+but only one context can be initialized and used at a time. Context properties
+must not be directly accessed or manipulated by the application.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc device instantiation start
+   :end-before: doc device instantiation end
+
+Your USB device may have manufacturer, product, and serial number string
+descriptors. To instantiate these string descriptors, the application should
+use the appropriate :c:macro:`USBD_DESC_MANUFACTURER_DEFINE`,
+:c:macro:`USBD_DESC_PRODUCT_DEFINE`, and
+:c:macro:`USBD_DESC_SERIAL_NUMBER_DEFINE` macros. String descriptors also
+require a single instantiation of the language descriptor using the
+:c:macro:`USBD_DESC_LANG_DEFINE` macro.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc string instantiation start
+   :end-before: doc string instantiation end
+
+String descriptors must be added to the device context at runtime before
+initializing the USB device with :c:func:`usbd_add_descriptor`.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc add string descriptor start
+   :end-before: doc add string descriptor end
```

Comment:
here and the other `literalinclude`s as well (even the ones that don't need it, IMO) -- this makes the code left aligned even when the included section happens to be indented.


```suggestion
.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
   :language: c
   :dedent:
   :start-after: doc add string descriptor start
   :end-before: doc add string descriptor end
```

This will make the below flush left

<img width="837" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/add98898-2ecd-4a2b-99a9-ea471c3fbf59">